UIComponents/Sources/UIComponents/Components/Inputs/Content/TextFieldContainable.swift

```diff
@@ -0,0 +1,190 @@
+import UIKit
+
+public protocol TextFieldContentViewChangesHandable: AnyObject {
+    var shouldBeginEditing: ((TextFieldContainable) -> Bool)? { get set }
+    var didBeginEditing: ((TextFieldContainable) -> Void)? { get set }
+    var shouldEndEditing: ((TextFieldContainable) -> Bool)? { get set }
+    var didEndEditing: ((TextFieldContainable) -> Void)? { get set }
+    var shouldChangeCharacters: ((TextFieldContainable, NSRange, String) -> Bool)? { get set }
+    var didChange: ((TextFieldContainable) -> Void)? { get set }
+    var shouldClear: ((TextFieldContainable) -> Bool)? { get set }
+    var shouldReturn: ((TextFieldContainable) -> Bool)? { get set }
+}
+
+public protocol TextFieldContainable: UIView, TextFieldContentViewChangesHandable {
+    var text: String? { get set }
+    var placeholder: String? { get set }
+    var attributedText: NSAttributedString? { get set }
+    var selectedTextRange: UITextRange? { get set }
+    var beginningOfDocument: UITextPosition { get }
+    var keyboardAppearance: UIKeyboardAppearance { get set }
+    var autocorrectionType: UITextAutocorrectionType { get set }
+    var keyboardType: UIKeyboardType { get set }
+    var returnKeyType: UIReturnKeyType { get set }
+    var autocapitalizationType: UITextAutocapitalizationType { get set }
+    var spellCheckingType: UITextSpellCheckingType { get set }
+    var isSecureTextEntry: Bool { get set }
+
+    var textColor: UIColor? { get set }
+    var font: UIFont? { get set }
+
+    var inputView: UIView? { get set }
+
+    func position(from: UITextPosition, offset: Int) -> UITextPosition?
+    func setCursorLocation(_ location: Int)
+}
+
+public final class TextFieldSingleLineView: UITextField, TextFieldContainable, UITextFieldDelegate {
```

Comment:
BTW, maybe `SingleLineTextFieldView` would be better?

Reply:
I decided to name it in such way due to fact we could clearly identify belonging of this UI component. Also, we can easily group such UI components using such naming strategy.